src/standalone/Soil/rre.jl

```diff
@@ -118,6 +122,19 @@ function make_update_boundary_fluxes(model::RichardsModel)
             p,
             t,
         )
+
+        # If p.soil.dfluxBCdY is present, update it
+        if haskey(p.soil, :dfluxBCdY)
```

Comment:
we have defined a default version of `set_dfluxBCdY!` which does `nothing` if `model::AbstractModel`. We have a more specific method defined below. Do we still need the if statement? or can we remove that, and call set_dfluxBCdY!, but in all cases but this one, it does nothing?

Reply:
Good point! `set_dfluxBCdY!` has to dispatch on the model type and also the boundary condition type to only update the cache for the combination of RichardsModel with a MoistureStateBC at the top boundary. But that's what the non-nothing method dispatches off of so I think we can just remove the if statement